spec/tapioca/compilers/dsl/state_machines_spec.rb

```diff
@@ -0,0 +1,306 @@
+      RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it("generate RBI for a classe with state_machine, verify path helper method only ") do
+      content = <<~RUBY
+        class Vehicle
+          attr_accessor :seatbelt_on, :time_used, :auto_shop_busy
+
+          state_machine :state, initial: :parked do
+            before_transition parked: any - :parked, do: :put_on_seatbelt
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        sig { params(args: T.untyped).returns(T::Array[::StateMachines::Transition]) }
+          def state_paths(*args); end
+      RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it(" generate RBI for a class with state_machine, verify helper methods start with_ and without") do
+      content = <<~RUBY
+
+        module CustomAttributeIntegration
+          include StateMachines::Integrations::Base
+          def self.integration_name
+            :custom_attribute
+          end
+
+          def create_with_scope(_name)
+            -> {}
+          end
+          def create_without_scope(_name)
+            -> {}
+          end
+        end
+        StateMachines::Integrations.register(CustomAttributeIntegration)
+        class Vehicle
+          state_machine :state, integration: :custom_attribute
+        end
+      RUBY
+
+      expected = <<~RUBY
+        sig { params(states: T.any(String, Symbol)).returns(T.untyped) }
+          def with_state(*states); end
+
+          sig { params(states: T.any(String, Symbol)).returns(T.untyped) }
+          def with_states(*states); end
+
+          sig { params(states: T.any(String, Symbol)).returns(T.untyped) }
+          def without_state(*states); end
+
+          sig { params(states: T.any(String, Symbol)).returns(T.untyped) }
+          def without_states(*states); end
+      RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it("generate RBI for a class with state_machine, verify helper method for machine.action ") do
+      content = <<~RUBY
+
+        module CustomAttributeIntegration
+          include StateMachines::Integrations::Base
+          def self.integration_name
+            :custom_attribute
+          end
+          @defaults = { action: :save, use_transactions: false }
+        end
+        StateMachines::Integrations.register(CustomAttributeIntegration)
+        class Vehicle
+          state_machine :state, integration: :custom_attribute
+        end
+      RUBY
+
+      expected = <<~RUBY
+        sig { returns(T.nilable(Symbol)) }
+          def state_event; end
+
+          sig { params(value: T.any(String, Symbol)).returns(T.any(String, Symbol)) }
+          def state_event=(value); end
+
+          sig { returns(T.nilable(::StateMachines::Transition)) }
+          def state_event_transition; end
+
+          sig { params(value: ::StateMachines::Transition).returns(::StateMachines::Transition) }
+          def state_event_transition=(value); end
```

Comment:
Can we fix the indentation here please?

Reply:
This is actually intentional! it would fail if we fix the indentation . I think it is the `<<~RUBY` operation that doesn't take the indentation as the space for the first line somehow.